### PR TITLE
block/aoe: initial server tests

### PR DESCRIPTION
**Pull request by mdlayher**
This is a big one.  I tried to set up some kind of sane testing framework for the AoE server.  After some trial and error, this is what I came up with.

This test, and others, will essentially verify that for a given AoE request, an expected AoE response is provided.

It's definitely not the prettiest piece of code I've ever written, and I'd be happy to hear any feedback you all may have about making it nicer.

### CODE REVIEW

File: block/aoe/aoe_test.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	serverRead := bytes.NewBuffer(nil)
+
+	bp := &bufPacketConn{
+		rb: serverWrite,
+		wb: serverRead,
+	}
+
+	iface := &Interface{
+		Interface: &net.Interface{
+			Name: "eth0",
```
Comment (mischief) [inline block/aoe/aoe_test.go]: can we just use lo?

Comment (mdlayher) [inline block/aoe/aoe_test.go]: Sure, but I don't think this value actually matters at all in our code anyway.

File: block/aoe/aoe_test.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	return resh
+}
+
+// testAoEServer sets up a Torus AoE server and returns:
+//  - an io.Reader which contains the server's response to a request
+//  - an io.Writer which can be used to issue a request to the server
+//  - a closure which can be used to run the server for a single
+//    request/response cycle
+//  - a closure which can be used to tear down all server resources
+func testAoEServer(t *testing.T) (response io.Reader, request io.Writer, run func() error, done func()) {
```
Comment (philips) [inline block/aoe/aoe_test.go]: you call it run here and start below, any reason?

**Comment by philips** [comment]
seems reasonable

File: block/aoe/aoe_test.go (hunk 2 above)
Comment (mdlayher) [inline block/aoe/aoe_test.go]: Nope, just a slight oversight on my part!  Updated.

**Pull request merged**